### PR TITLE
Transfer to spring-boot-3.1.2 and oauth2 authorization

### PR DESCRIPTION
**Pull request by michael200kg**
Transfer to a Spring boot 3 
Add OAuth2 authorization
Made some code clean up

### CODE REVIEW

File: ui/src/main/java/com/epam/eco/kafkamanager/ui/GlobalModelEnrichingInterceptor.java [hunk 1]
```diff
@@ -55,9 +62,23 @@ public void postHandle(
         }
 
         modelAndView.addObject(ATTR_BUILD_INFO, BuildInfo.instance());
-        modelAndView.addObject(
-                ATTR_PRINCIPAL,
-                SecurityContextHolder.getContext().getAuthentication());
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if(nonNull(SecurityContextHolder.getContext().getAuthentication())) {
```
Comment (YevgeniyYP) [inline ui/src/main/java/com/epam/eco/kafkamanager/ui/GlobalModelEnrichingInterceptor.java:68]: Check `authentication`  variable on null instead

File: ui/src/main/java/com/epam/eco/kafkamanager/ui/GlobalModelEnrichingInterceptor.java [hunk 2]
```diff
@@ -15,25 +15,33 @@
...
 import com.epam.eco.kafkamanager.ui.config.KafkaManagerUiProperties;
 
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
 /**
  * @author Andrei_Tytsik
  */
-public class GlobalModelEnrichingInterceptor extends HandlerInterceptorAdapter {
+public class GlobalModelEnrichingInterceptor implements HandlerInterceptor {
```
Comment (YevgeniyYP) [inline ui/src/main/java/com/epam/eco/kafkamanager/ui/GlobalModelEnrichingInterceptor.java:39]: Extending `WebRequestHandlerInterceptorAdapter` is not mandatory here?

File: ui/src/main/java/com/epam/eco/kafkamanager/ui/config/BaseWebSecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler;
+
+import java.util.Collections;
+
+/**
+ * @author Mikhail_Vershkov
+ */
+public class BaseWebSecurityConfig {
```
Comment (YevgeniyYP) [inline ui/src/main/java/com/epam/eco/kafkamanager/ui/config/BaseWebSecurityConfig.java:34]: Can it be abstract and `commonHttpConfig()` protected?

File: ui/src/main/java/com/epam/eco/kafkamanager/ui/config/KeycloakBaseWebSecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            }
+            return mappedAuthorities;
+        };
+    }
+
+    private void addExtractedRoles(Map<String, Object> resourceAccess,
+                                   OidcUserAuthority oidcUserAuthority,
+                                   Set<GrantedAuthority> mappedAuthorities) {
+        Collection<String> roles = resourceAccess.values().stream()
+                .flatMap(x -> ((Map<String, Object>) x).entrySet().stream())
```
Comment (YevgeniyYP) [inline ui/src/main/java/com/epam/eco/kafkamanager/ui/config/KeycloakBaseWebSecurityConfig.java:101]: Type casting looks harmful without pre-checking

**Review by YevgeniyYP — CHANGES_REQUESTED** [review]

File: commons/src/test/java/com/epam/eco/kafkamanager/KafkaSchemaIdAwareAvroDeserializerTest.java [hunk 5]
```diff
@@ -12,10 +12,10 @@
 
 public class KafkaSchemaIdAwareAvroDeserializerTest {
 
-    private final byte[] BYTES_SHORT = new byte[4];
-    private final byte[] BYTES_NORMAL = new byte[20];
-    private final byte BYTE_FILL = 1;
-    private final byte BYTE_ZERO = 0;
+    private final static byte[] BYTES_SHORT = new byte[4];
```
[the remark this answers is not in the document]
Comment (michael200kg) [reply commons/src/test/java/com/epam/eco/kafkamanager/KafkaSchemaIdAwareAvroDeserializerTest.java:15]: just memory efficiency nothing more

File: commons/src/test/java/com/epam/eco/kafkamanager/exec/AbstractAsyncStatefullTaskExecutorTest.java [hunk 6]
```diff
@@ -155,7 +156,7 @@ public void testLastFreshResultIsResolvedPriorToRunningNewTask() throws Exceptio
 
     private class SimpleParrotTaskExecutor extends AbstractAsyncStatefullTaskExecutor<String, String> {
 
-        private Collection<String> phrases;
+        private final Collection<String> phrases;
         private Iterator<String> iterator;
```
[the remark this answers is not in the document]
Comment (michael200kg) [reply commons/src/test/java/com/epam/eco/kafkamanager/exec/AbstractAsyncStatefullTaskExecutorTest.java:160]: Above in the code:
iterator = phrases.iterator();

**Review by YevgeniyYP — APPROVED** [review]

**Pull request merged**